src/lib/core/a11y/interactivity-checker.spec.ts

```diff
@@ -234,63 +237,274 @@ describe('InteractivityChecker', () => {
+      it('should return true for div and span with tabindex == 0', () => {
+        let elements = createElements('div', 'span');
 
-      elements.forEach(el => el.setAttribute('tabindex', '-1'));
-      appendElements(elements);
+        elements.forEach(el => el.setAttribute('tabindex', '0'));
+        appendElements(elements);
 
-      elements.forEach(el => {
-        expect(checker.isTabbable(el))
-            .toBe(false, `Expected <${el.nodeName} tabindex="-1"> not to be tabbable`);
+        elements.forEach(el => {
+          expect(checker.isFocusable(el))
+          .toBe(true, `Expected <${el.nodeName} tabindex="0"> to be focusable`);
+        });
       });
-    });
 
-    it('should return true for div and span with tabindex == 0', () => {
-      let elements = createElements('div', 'span');
+      it('should return false for native form controls and anchor with tabindex == -1', () => {
+        let elements = createElements('input', 'textarea', 'select', 'button', 'a');
 
-      elements.forEach(el => el.setAttribute('tabindex', '0'));
-      appendElements(elements);
+        elements.forEach(el => el.setAttribute('tabindex', '-1'));
+        appendElements(elements);
 
-      elements.forEach(el => {
-        expect(checker.isTabbable(el))
-            .toBe(true, `Expected <${el.nodeName} tabindex="0"> to be tabbable`);
+        elements.forEach(el => {
+          expect(checker.isTabbable(el))
+          .toBe(false, `Expected <${el.nodeName} tabindex="-1"> not to be tabbable`);
+        });
       });
-    });
+
+      it('should return true for div and span with tabindex == 0', () => {
+        let elements = createElements('div', 'span');
+
+        elements.forEach(el => el.setAttribute('tabindex', '0'));
+        appendElements(elements);
+
+        elements.forEach(el => {
+          expect(checker.isTabbable(el))
+          .toBe(true, `Expected <${el.nodeName} tabindex="0"> to be tabbable`);
+        });
+      });
+
+      it('should respect the inherited tabindex inside of frame elements', () => {
+        let iframe = createFromTemplate('<iframe>', true) as HTMLFrameElement;
+        let button = createFromTemplate('<button tabindex="0">Not Tabbable</button>');
+
+        appendElements([iframe]);
+
+        iframe.tabIndex = -1;
+        iframe.contentDocument.body.appendChild(button);
+
+        expect(checker.isTabbable(iframe)).toBe(false);
+        expect(checker.isTabbable(button)).toBe(false);
+
+        iframe.tabIndex = null;
+
+        expect(checker.isTabbable(iframe)).toBe(false);
+        expect(checker.isTabbable(button)).toBe(true);
+      });
+
+      it('should mark elements which are contentEditable as tabbable', async(() => {
+        let editableEl = createFromTemplate('<div contenteditable="true">', true);
+
+        // Wait one tick, because the browser takes some time to update the tabIndex
+        // according to the contentEditable attribute.
+        setTimeout(() => {
```

Comment:
interesting, can we just change the timeout to 0 then, bumping to the end of the queue with no delay should be enough